### PR TITLE
test_all.sh: configure fork test timeout

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
Example output:
```
$ ./test_all.sh 2>/dev/null
PASS abfrmblr
PASS AbstractKamen
PASS ags313
PASS alesj
PASS algirdasrascius
PASS anandmattikopp
PASS arjenvaneerde
FAIL armandino
TIME artpar
PASS artsiomkorzun
PASS asun
PASS baseline
PASS berry120
PASS bjhara
PASS charlibot
PASS coolmineman
PASS couragelee
PASS criccomini
PASS davecom
PASS davery22
PASS ddimtirov
PASS deemkeen
PASS ebarlas
PASS entangled90
FAIL fatroom
PASS felix19350
PASS filiphr
PASS flippingbits
PASS fragmede
PASS gabrielreid
PASS gamlerhart
PASS gnmathur
PASS hchiorean
FAIL hundredwatt
PASS imrafaelmerino
PASS isolgpus
PASS itaske
PASS iziamos
PASS japplis
PASS jgrateron
PASS jotschi
PASS kevinmcmurtrie
PASS kgeri
PASS khmarbaise
PASS kuduwa-keshavram
PASS kumarsaurav123
FAIL lawrey
PASS maeda6uiui
PASS MeanderingProgrammer
PASS merykitty
FAIL moysesb
PASS mudit-saxena
PASS obourgain
PASS padreati
PASS palmr
PASS pedestrianlove
PASS phd3
PASS raipc
PASS rby
PASS richardstartin
FAIL roman-r-m
PASS royvanrijn
PASS rprabhu
PASS santanu
PASS seijikun
PASS semotpan
PASS spullara
PASS thomaswue
PASS truelive
FAIL twobiers
PASS Ujjwalbharti
PASS yavuztas
PASS yehwankim23
FAIL yemreinci
PASS zerninv
```

### CODE REVIEW

File: test_all.sh [hunk 1]
```diff
@@ -33,14 +33,35 @@ if [ "$INPUT" = "-h" ] || [ "$#" -gt 1 ]; then
...
+
 for impl in $(ls calculate_average_*.sh | sort); do
   noext="${impl%%.sh}"
   fork=${noext##calculate_average_}
 
-  if output=$(./test.sh "$fork" "$INPUT" 2>&1); then
+  # ./test.sh calls ./prepare_$fork.sh e.g. to build native image
+  # which may take some time.
+  # Here we run it upfront, assuming that prepare result is cached
+  # to avoid timeout due to long preparation.
```
Comment (AlexanderYastrebov) [inline test_all.sh:50]: It should cache anyways until new `mvnw clean verify`. Currently there are two submissions that build native image and they put it into target/ which works as expeceted.

Of course second run for test_all.sh completes faster due to cached native image.

**Comment by gunnarmorling** [comment]
Nice one. While at it, can we add the colored shell output similar to evaluate2.sh, rendering `PASS` in green and the others in red?

**Comment by AlexanderYastrebov** [comment]
> can we add the colored shell output

Not a fan of it but added.

**Comment by gunnarmorling** [comment]
Awesome, thx!

**Pull request merged**